backend-api/app/repository/sql/download.sql

```diff
@@ -1,358 +1,195 @@
-WITH deduplicated_family_slugs AS (
-        SELECT DISTINCT
-            ON (slug.family_import_id) slug.family_import_id,
-            slug.created,
-            slug.name
-        FROM
-            (
-                SELECT
-                    slug.family_import_id,
-                    COUNT(*) AS count
-                FROM
-                    slug
-                WHERE
-                    slug.family_import_id IS NOT NULL
-                GROUP BY
-                    slug.family_import_id
-                HAVING
-                    COUNT(*) > 1
-            ) AS duplicates
-        LEFT JOIN slug ON duplicates.family_import_id = slug.family_import_id
-        ORDER BY
-            slug.family_import_id DESC,
-            slug.created DESC,
-            slug.ctid DESC
-    ),
-
-unique_family_slugs AS (
-        SELECT DISTINCT
-            ON (slug.family_import_id) slug.family_import_id,
-            slug.created,
-            slug.name
-        FROM
-            (
-                SELECT
-                    slug.family_import_id,
-                    COUNT(*) AS count
-                FROM
-                    slug
-                WHERE
-                    slug.family_import_id IS NOT NULL
-                GROUP BY
-                    slug.family_import_id
-                HAVING
-                    COUNT(*) = 1
-            ) AS non_duplicates
-        LEFT JOIN
-            slug
-            ON non_duplicates.family_import_id = slug.family_import_id
-        ORDER BY
-            slug.family_import_id DESC,
-            slug.created DESC,
-            slug.ctid DESC
-    ),
-
-most_recent_family_slugs AS (
-        SELECT
-            deduplicated_family_slugs.family_import_id,
-            deduplicated_family_slugs.created,
-            deduplicated_family_slugs.name
-        FROM
-            deduplicated_family_slugs
-        UNION ALL
-        SELECT
-            unique_family_slugs.family_import_id,
-            unique_family_slugs.created,
-            unique_family_slugs.name
-        FROM
-            unique_family_slugs
-        ORDER BY
-            family_import_id DESC,
-            created DESC
-    ),
+-- Optimised version with improved performance
+WITH
+-- Single CTE for all slug deduplication (family and document)
+recent_slugs AS (
+    SELECT DISTINCT ON (
+        COALESCE(family_import_id::TEXT, '')
+        || '|'
+        || COALESCE(family_document_import_id::TEXT, '')
```

Comment:
Nonblocking suggestion: COACLESCE can be a bit weird with indexes. Did you run explain analyze (when not under massive time constraint i guess!) to work out if performance is all good?